packages/resource-deployment/scripts/pool-startup/pull-image-from-container-registry.ps1

```diff
@@ -115,13 +115,13 @@ if ([string]::IsNullOrEmpty($global:keyvault)) {
     exitWithUsageInfo
 }
 
-try {
-    getCurrentUserDetails
-    loginToAzure
-    grantUserAccessToKeyVault
-    loginToContainerRegistry
-    pullDockerImages
-}
-catch {
+trap {
```

Comment:
The previous implementation looked like revokeUserAccessToKeyVault was a cleanup step performed if there was an exception in any of the preceding steps; now it is performed prior to executing the real work. What's the significance of moving this before that work?